presto-cache/src/main/java/com/facebook/presto/cache/alluxio/AlluxioCachingClientFileSystem.java

```diff
@@ -0,0 +1,257 @@
+            throws IOException
+    {
+        fileSystem.delete(toPath(alluxioURI), options.getRecursive());
+    }
+
+    @Override
+    public boolean exists(AlluxioURI alluxioURI, ExistsPOptions options)
+            throws IOException
+    {
+        return fileSystem.exists(toPath(alluxioURI));
+    }
+
+    @Override
+    public AlluxioConfiguration getConf()
+    {
+        return alluxioConfiguration;
+    }
+
+    @Override
+    public URIStatus getStatus(AlluxioURI path, GetStatusPOptions options)
+            throws IOException
+    {
+        return toUriStatus(fileSystem.getFileStatus(toPath(path)));
+    }
+
+    @Override
+    public List<URIStatus> listStatus(AlluxioURI path, ListStatusPOptions options)
+            throws IOException
+    {
+        List<URIStatus> statuses = Arrays.stream(fileSystem.listStatus(toPath(path))).map(this::toUriStatus).collect(Collectors.toList());
+        return statuses;
+    }
+
+    @Override
+    public boolean isClosed()
+    {
+        throw new UnsupportedOperationException("isClosed is not supported");
+    }
+
+    @Override
+    public void free(AlluxioURI path, FreePOptions options)
+    {
+        throw new UnsupportedOperationException("Free is not supported.");
+    }
+
+    private BlockLocationInfo toBlockLocationInfo(BlockLocation blockLocation)
+    {
+        throw new UnsupportedOperationException("toBlockLocationInfo is not supported.");
+    }
+
+    @Override
+    public List<BlockLocationInfo> getBlockLocations(AlluxioURI alluxioURI)
+            throws IOException
+    {
+        Path path = toPath(alluxioURI);
+        BlockLocation[] fileBlockLocations = fileSystem.getFileBlockLocations(path, 0, path.toString().length());
+        Arrays.stream(fileBlockLocations).map(this::toBlockLocationInfo).collect(Collectors.toList());
+
+        throw new UnsupportedOperationException("GetBlockLocations is not supported.");
+    }
+
+    @Override
+    public void mount(AlluxioURI alluxioPath, AlluxioURI ufsPath, MountPOptions options)
+    {
+        throw new UnsupportedOperationException("Mount is not supported.");
+    }
+
+    @Override
+    public void updateMount(AlluxioURI alluxioPath, MountPOptions options)
+    {
+        throw new UnsupportedOperationException("UpdateMount is not supported.");
+    }
+
+    @Override
+    public Map<String, MountPointInfo> getMountTable()
+    {
+        throw new UnsupportedOperationException("GetMountTable is not supported.");
+    }
+
+    @Override
+    public List<SyncPointInfo> getSyncPathList()
+    {
+        throw new UnsupportedOperationException("GetSyncPathList is not supported.");
+    }
+
+    @Override
+    public void persist(AlluxioURI path, ScheduleAsyncPersistencePOptions options)
+    {
+        throw new UnsupportedOperationException("Persist is not supported.");
+    }
+
+    @Override
+    public void rename(AlluxioURI src, AlluxioURI dst, RenamePOptions options)
```

Comment:
We don't use abbreviation in our codebase; so `source`, `destination`, and many other variables in this class

Reply:
Updated all occurrences.